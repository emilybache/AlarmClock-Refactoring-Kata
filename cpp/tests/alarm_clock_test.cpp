#include <AlarmClock.h>
#include <climits>

#include "doctest.h"
#include "ApprovalTests.hpp"

TEST_CASE ("no alarms are set") {

    unsigned int now_sec = 100; // 100 seconds since the epoch
    unsigned long min_value_ms = INT_MAX;

    auto *config = new alarm_config();
    config->timers = new timers();
    config->timers->duration = new duration();

    how_long_until_the_next_alarm(config, now_sec, &min_value_ms);
    
    std::ostringstream result;
    result << "no alarms set: " << min_value_ms;
    std::string to_verify = result.str();

    Approvals::verify(to_verify);
}
