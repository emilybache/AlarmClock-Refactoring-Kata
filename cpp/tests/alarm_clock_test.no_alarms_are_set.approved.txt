no alarms set: 2147483647
